src/main/java/org/wso2/carbon/inbound/kafka/KafkaMessageConsumer.java

```diff
@@ -0,0 +1,444 @@
+            }
+
+            if (properties.getProperty(KafkaConstants.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM) != null) {
+                kafkaProperties.put(KafkaConstants.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM,
+                        properties.getProperty(KafkaConstants.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM));
+            }
+
+            if (properties.getProperty(KafkaConstants.SSL_CIPHER_SUITES) != null) {
+                kafkaProperties.put(KafkaConstants.SSL_CIPHER_SUITES,
+                        properties.getProperty(KafkaConstants.SSL_CIPHER_SUITES));
+            }
+
+            if (properties.getProperty(KafkaConstants.SSL_KEYMANAGER_ALGORITHM) != null) {
+                kafkaProperties.put(KafkaConstants.SSL_KEYMANAGER_ALGORITHM,
+                        properties.getProperty(KafkaConstants.SSL_KEYMANAGER_ALGORITHM));
+            }
+
+            if (properties.getProperty(KafkaConstants.SSL_SECURE_RANDOM_IMPLEMENTATION) != null) {
+                kafkaProperties.put(KafkaConstants.SSL_SECURE_RANDOM_IMPLEMENTATION,
+                        properties.getProperty(KafkaConstants.SSL_SECURE_RANDOM_IMPLEMENTATION));
+            }
+
+            if (properties.getProperty(KafkaConstants.SSL_TRUSTMANAGER_ALGORITHM) != null) {
+                kafkaProperties.put(KafkaConstants.SSL_TRUSTMANAGER_ALGORITHM,
+                        properties.getProperty(KafkaConstants.SSL_TRUSTMANAGER_ALGORITHM));
+            }
+
+            consumer = new KafkaConsumer<>(kafkaProperties);
+        } catch (Exception e) {
+            throw new SynapseException("Failed to construct kafka consumer", e);
+        }
+
+        String[] topicsArray = topic.split(",");
+        consumer.subscribe(Arrays.asList(topicsArray));
+
+        try {
+            while (true) {
+                ConsumerRecords<byte[], byte[]> records = consumer.poll(Long.parseLong(pollTimeout));
+                for (ConsumerRecord record : records) {
+                    msgCtx = createMessageContext();
+                    msgCtx.setProperty(KafkaConstants.KAFKA_PARTITION_NO, record.partition());
+                    msgCtx.setProperty(KafkaConstants.KAFKA_MESSAGE_VALUE, record.value());
+                    msgCtx.setProperty(KafkaConstants.KAFKA_OFFSET, record.offset());
+                    msgCtx.setProperty(KafkaConstants.KAFKA_CHECKSUM, record.checksum());
+                    msgCtx.setProperty(KafkaConstants.KAFKA_TIMESTAMP, record.timestamp());
+                    msgCtx.setProperty(KafkaConstants.KAFKA_TIMESTAMP_TYPE, record.timestampType());
+                    msgCtx.setProperty(KafkaConstants.KAFKA_TOPIC, record.topic());
+                    msgCtx.setProperty(KafkaConstants.KAFKA_KEY, record.key());
+                    injectMessage(record.value().toString(), contentType);
+                }
+            }
+        } catch (WakeupException ex) {
+            log.error("Error while wakeup the consumer" + consumer);
```

Comment:
It is handled by kafka itself. If there any issue while wakeup the consumer we must need to close the consumer. I have tried this use case as stop the broker and stop the network as well and the consumer remains as it is. When we destroy the inbound only it catch that exception and create a new consumer with the new configuration.